updated the highlight

MbfloydIR opened the pull request:
## Summary
updated the highlight logic to also highlight children of the selected object
increased thickness of the highlight

## Subtasks Checklist

## Breaking Changes

## References
closes #_insert number here_

## QA Steps

MbfloydIR committed:
updated the highlight logic to also highlight children of the selected object

MichaelEstes commented on packages/spatial/src/renderer/components/HighlightComponent.ts:
```diff
@@ -26,23 +26,32 @@ Ethereal Engine. All Rights Reserved.
   /** @todo support multiple scenes */
   if (!hasComponent(Engine.instance.viewerEntity, RendererComponent)) return
 
+  let highlightObjects: Object3D[] = []
```
If you change this type to a Set<Object3D> it'll remove the need for the ts-ignore below

MbfloydIR committed:
removed the ts ignore

MbfloydIR committed:
Updated var name from "test"
Put the ts ignore back since it is still throwing an error "../spatial/src/renderer/components/HighlightComponent.ts(53,24): error TS2339: Property 'OutlineEffect' does not exist on type 'EffectComposer'."

MbfloydIR committed:
removed the ts ignore and changed the array to a set type

MichaelEstes committed:
Re-export postprocessing

HexaField commented on packages/spatial/src/renderer/components/PostProcessingComponent.tsx:
```diff
@@ -135,6 +137,9 @@ const PostProcessingReactor = (props: { entity: Entity; rendererEntity: Entity }
     // //always have the outline effect for the highlight selection
     const outlineEffect = new OutlineEffect(scene as Scene, camera.value as ArrayCamera, getState(HighlightState))
     outlineEffect.selectionLayer = ObjectLayers.HighlightEffect
+    outlineEffect.blur = true
```
we should use the values in `HighlightState` instead of hardcoding these, and fix those to be the settings we want

MbfloydIR replied:
implemented

MbfloydIR committed:
moved the hardcoded values to the highlight state

MbfloydIR committed:
Merge branch 'michael/highlight_updates' of https://github.com/EtherealEngine/etherealengine into michael/highlight_updates

* 'michael/highlight_updates' of https://github.com/EtherealEngine/etherealengine:
  Re-export postprocessing

HexaField reviewed (changes requested):
![Screenshot from 2024-07-19 08-05-01](https://github.com/user-attachments/assets/d457e3fe-ef48-45b8-accb-363947fdd02b)
we need to check if the effect composer exists or not

HexaField commented on packages/spatial/src/renderer/components/HighlightComponent.ts:
```diff
@@ -26,23 +26,31 @@ Ethereal Engine. All Rights Reserved.
+  for (const entity of highlightQuery()) {
+    traverseEntityNode(entity, (child, index) => {
+      if (!hasComponent(child, MeshComponent)) return
+      if (!hasComponent(child, GroupComponent)) return
```
we need to exclude non visible entities here too

MbfloydIR replied:
implemented

HexaField commented on packages/spatial/src/renderer/components/HighlightComponent.ts:
```diff
@@ -26,23 +26,31 @@ Ethereal Engine. All Rights Reserved.
-  getComponent(Engine.instance.viewerEntity, RendererComponent).effectComposer?.OutlineEffect?.selection.set(
-    highlightQuery().map((entity) => getComponent(entity, MeshComponent))
-  )
+  let highlightObjects = new Set<Object3D>()
```
this `let` should be a `const`

MbfloydIR replied:
Implemented

MbfloydIR committed:
Merge branch 'dev' into michael/highlight_updates

* dev:
  hiding the new and delete buttons for now till the we can do a full rework of materials as assets after phase 1 (#10656)
  Fix regression with instance startup (#10649)
  fix material renaming + IO (#10640)
  avoid crashing app when key not found (#10513)
  asset panel totalPages should be an integer (#10646)
  IR-3191-MT-INT-Studio-an-extra-window-is-called-when-saving-via-a-key-combination (#10643)
  Added migrate in vs debug config
  IR-3192-MT-INT-Studio-No-warning-notification-when-switching-between-unsaved-scenes (#10644)
  IR-2985-Point-Click-doesnt-work-with-Assets-Panel (#10612)
  Fixed issues with creating consumers (#10641)
  Fix hotkeys in Heirarchu Panel (#10632)
  IR-3172-Unable-to-change-Image-URL (#10642)
  IR-3186/3185/3184/3220 Fix material object reference loss, material library rerendering, fallback and UI improvements  (#10627)
  add tag to save material (#10634)
  IR-3118: Material thumbnail generation (#10626)
  fix ignore geometry to not delete all meshes (#10639)
  update addMediaNode (#10637)
  updated the material preview height so that the list of materials could still be seen when one is selected (#10638)

MbfloydIR committed:
added check for if the effect composer exists or not
and exclude non visible entities
changed let to be a const
